### ISSUE TITLE
`chan struct{}` triggers nested-structs error

### ISSUE THREAD

**Issue by Integralist** [mgechev/revive#744]
**Describe the bug**
`chan struct{}` triggers the nested-structs error.

**Expected behavior**
I expected this use case to be ignored.

**Desktop (please complete the following information):**
 - OS: macOS Monterey
 - Version of Go: 1.19

**Additional context**
I'm presuming this is a false positive bug and there's no idiomatic go way to work around it (apologies if there is and I wasn't aware).

**Comment by chavacava** [comment]
Hi @Integralist, thanks for filling the issue.
Does your issue refer to false-positives in cases like 
```go
type issue744 struct {
	c chan struct{}
	// ...other fields...
}
```
?
If yes, the branch [fix/744](https://github.com/mgechev/revive/tree/fix/744) should fix the problem. Could you confirm?

**Comment by Integralist** [comment]
Hi @chavacava correct! I've noticed it specifically here where we have a struct field of `chan struct{}`:

https://github.com/fastly/cli/blob/c4a65100aeacabc740e5032c81034c15beed6ab3/pkg/commands/logtail/root.go#L30-L42
